### PR TITLE
Fix Webpack i18n Extractions 

### LINKED ISSUE #3772: Fix eejs console errors
Using the latest release (and master) versions I'm getting this in the console:

Uncaught TypeError: Cannot set properties of undefined (setting 'eejs')
    at eejs-core.c1e0cebdd6b75e1f4751.dist.js?ver=c1e0cebdd6b75e1f47518e3fa423376e:1:7672
    at eejs-core.c1e0cebdd6b75e1f4751.dist.js?ver=c1e0cebdd6b75e1f47518e3fa423376e:1:7675
(index):443 Uncaught ReferenceError: eejs is not defined
    at (index):443:1

To test just load (apparently) any page, I used the home page of a site.

Look in Dev tools console.

Reports:
https://eventespresso.com/topic/ee-js-errors-in-console/
https://eventespresso.com/topic/console-errors-ee4/
https://eventespresso.com/topic/eejs-not-defined-error/

### PR DESCRIPTION
**Pull request by tn3rb**
this PR:
- fixes #3772
- refactors our `wpi18nExtractor` class to work with Webpack 5
- refactors our webpack config
- updates project dependencies

# @ REVIEWERS => IMPORTANT 
when reviewing, ignore all files in the `/assets/dist/` folder as those are assets built by webpack

### CODE REVIEW

File: bin/i18n-map-extractor.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 const recast = require( 'recast' );
```
Comment (tn3rb) [inline bin/i18n-map-extractor.js:1]: # START REVIEW HERE

**Review by joaovictorangeline — APPROVED** [review]
Amazing job! 💯

**Comment by joaovictorangeline** [comment]
Tested it and no console errors anymore.
CleanShot 2022-05-13 at 11 45 40@2x
CleanShot 2022-05-13 at 11 46 10@2x

**Comment by Pebblo** [comment]
@tn3rb this does indeed fix the issue mentioned but unit tests are filing, can you check those plz?

**Comment by tn3rb** [comment]
@Pebblo 
tests are all fixed now. 
plz ignore the failed JS lint tests as that code is removed in dev branch so I'm not going to waste time trying to make the code look pretty

**Pull request merged**